libraries/AP_Motors/AP_Motors_Thrust_Linearization.cpp

```diff
@@ -5,6 +5,57 @@
 #include <AP_Baro/AP_Baro.h>
 #include <AP_Vehicle/AP_Vehicle_Type.h>
 
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo Thrust_Linearization::var_info[] = {
+
+    // @Param: THST_EXPO
+    // @DisplayName: Thrust Curve Expo
+    // @Description: motor thrust curve exponent (0.0 for linear to 1.0 for second order curve)
+    // @Range: -1 1
+    // @User: Standard
+    AP_GROUPINFO("THST_EXPO", 1, Thrust_Linearization,  curve_expo, 0.55),
+
+    // @Param: SPIN_MIN
+    // @DisplayName: Motor Spin minimum
+    // @Description: Point at which the thrust starts expressed as a number from 0 to 1 in the entire output range.  Should be higher than MOT_SPIN_ARM.
+    // @Values: 0.0:Low, 0.15:Default, 0.3:High
+    // @User: Standard
+    AP_GROUPINFO("SPIN_MIN", 2, Thrust_Linearization,  spin_min, 0.15),
+
+    // @Param: SPIN_MAX
+    // @DisplayName: Motor Spin maximum
+    // @Description: Point at which the thrust saturates expressed as a number from 0 to 1 in the entire output range
+    // @Values: 0.9:Low, 0.95:Default, 1.0:High
+    // @User: Standard
+    AP_GROUPINFO("SPIN_MAX", 3, Thrust_Linearization,  spin_max, 0.95),
+
+    // @Param: BAT_IDX
+    // @DisplayName: Battery compensation index
+    // @Description: Which battery monitor should be used for doing compensation
+    // @Values: 0:First battery, 1:Second battery
+    // @User: Standard
+    AP_GROUPINFO("BAT_IDX", 4, Thrust_Linearization, batt_idx, 0),
+
+    // @Param: BAT_V_MAX
+    // @DisplayName: Battery voltage compensation maximum voltage
+    // @Description: Battery voltage compensation maximum voltage (voltage above this will have no additional scaling effect on thrust).  Recommend 4.2 * cell count, 0 = Disabled
+    // @Range: 6 53
+    // @Units: V
+    // @User: Standard
+    AP_GROUPINFO("BAT_V_MAX", 5, Thrust_Linearization, batt_voltage_max, 0),
+
+    // @Param: BAT_V_MIN
+    // @DisplayName: Battery voltage compensation minimum voltage
+    // @Description: Battery voltage compensation minimum voltage (voltage below this will have no additional scaling effect on thrust).  Recommend 3.3 * cell count, 0 = Disabled
+    // @Range: 6 42
+    // @Units: V
+    // @User: Standard
+    AP_GROUPINFO("BAT_V_MIN", 6, Thrust_Linearization, batt_voltage_min, 0),
+
+    AP_GROUPEND
+};
+
```

Comment:
@IamPete1 i moved the parameter group into the thrust linearization class. Does that mess up the way you are adding these parameters for multi’s.

Reply:
Everything looks good with heli parameters using this new table in the Thrust linearization class